core/textinput/src/textinput/Editor.cpp

```diff
@@ -430,12 +464,20 @@ namespace textinput {
         return kPRSuccess;
       case kCmdReverseSearch:
         PushUndo();
-        fMode = kHistSearchMode;
-        fSearch.clear();
+        if (fMode != kHistFwdSearchMode) fSearch.clear();
```

Comment:
We can just use `fSearch.clear();`, can't we?

I think we don't reach here if `fMode == kHist{Rev,Fwd}SearchMode` (see lines 287-289 and lines 298-300).

Reply:
good point